game/src/game/components/RaycastComponent.cpp

```diff
@@ -19,43 +20,53 @@ void RaycastComponent::OnInit(const ServiceProvider& service_provider)
 {
     Log::info("RaycastComponent - Init");
 
-    // dependencies
-    transform_ = &GetEntity().GetComponent<Transform>();
-    input_service_ = &service_provider.GetService<InputService>();
+    // service dependencies
     physics_service_ = &service_provider.GetService<PhysicsService>();
 
-    // events
+    // component dependencies
+    transform_ = &GetEntity().GetComponent<Transform>();
+
+    // ok so i realized vehicles don't have box rigid bodies
+    // box_rigid_body_ = &GetEntity().GetComponent<BoxRigidBody>();
+    // self_shape_ = box_rigid_body_->GetShape();
+
     GetEventBus().Subscribe<OnUpdateEvent>(this);
 }
 
 void RaycastComponent::OnUpdate(const Timestep& delta_time)
 {
-    // mouse right-click to raycast
-    if (!input_service_->IsMouseButtonPressed(GLFW_MOUSE_BUTTON_2))
-        return;
+    // right-click to raycast
+    // if (!input_service_->IsMouseButtonPressed(GLFW_MOUSE_BUTTON_2))
+    //     return;
 
-    /*
-    note: rn object hits itself with its own cast
-    when i change direction to upDirection and shoot immediately
-    i get a hit for a split second and then never again lol
-    */
-    glm::vec3 origin = transform_->GetPosition();
+    // origin and direction of the raycast from this entity
     glm::vec3 direction = transform_->GetForwardDirection();
+    glm::vec3 origin = transform_->GetPosition();
+
+    std::cout << "forward direction: " << direction << std::endl;
 
     // no raycast data == no hit
     if (!physics_service_->Raycast(origin, direction).has_value())
         return;
 
-    // get data from raycast hit:
+    // get the data from the raycast hit
     RaycastData raycast = physics_service_->Raycast(origin, direction).value();
+
+    // physx::PxActor* self_actor = self_shape_->getActor();
+    physx::PxActor* actor = raycast.actor;
+
+    // raycast ignores own entity
+    // if (actor == self_actor)
+    // return;
+
```

Comment:
If these aren't used anymore we might as well just delete it

Reply:
oh good catch, i just missed them !